internal/util/lock/lock.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2024 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lock
+
+import (
+	"context"
+	"fmt"
+	"syscall"
+	"time"
+
+	"github.com/ceph/go-ceph/rados"
+
+	"github.com/ceph/ceph-csi/internal/util/log"
+)
+
+// IOCtxLock provides methods for acquiring and releasing exclusive locks on a volume.
+// using rados IO context locks.
+type IOCtxLock interface {
+	LockExclusive(ctx context.Context) error
+	Unlock(ctx context.Context)
+}
+
+type lock struct {
+	volID      string
+	lockName   string
+	lockDesc   string
+	lockCookie string
+	flags      *byte
+	timeout    time.Duration
+	ioctx      *rados.IOContext
+}
+
+// NewLock returns `lock` type that implements the IOCtxLock interface.
+func NewLock(
+	ioctx *rados.IOContext,
+	volID string,
+	lockName string,
+	lockCookie string,
+	lockDesc string,
+	timeout time.Duration,
+	flags *byte,
```

Comment:
flags is a very undescriptive parameter, and always is passed as `(*byte) 0`. Just drop it and pass the a pointer to a 0 byte in `LockExclusive`.

Reply:
Done.